### PR TITLE
GH-2 Initial project setup

### PR DESCRIPTION
**Pull request by rameshmalla**
#3

### CODE REVIEW

File: src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class MissionController(
+        private val objectMapper: ObjectMapper
```
Comment (prashanth-92) [inline src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt:17]: is objectmapper required?

File: src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return null
+    }
+
+    @PostMapping("/teams/{team-id}/missions", produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    fun createMission(@PathVariable("team-id") teamId: String, @RequestBody mission: Mission): MissionDetail {
+        return MissionDetail(id = "1222", mission = mission)
+    }
+
+    @PutMapping("/teams/{team-id}/missions/{mission-id}", produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    fun updateMission(@PathVariable("team-id") teamId: String, @PathVariable("mission-id") missionId: String, @RequestBody mission: Mission): MissionDetail {
```
Comment (prashanth-92) [inline src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt:36]: is @RequestBody required?

File: src/main/resources/db/migration/V001__CREATE_MISSIONS_TABLE.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  is_explored text not null,
+  status text,
+  sprint text,
+  team_id text not null,
+  critical text,
+  comments text[],
+  created_at timestamp without time zone,
+  updated_at timestamp without time zone,
+  primary key (id,team_id)
+);
```
Comment (prashanth-92) [inline src/main/resources/db/migration/V001__CREATE_MISSIONS_TABLE.sql:18]: Do we need an updated_by also for audit?

File: src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt (hunk 1 above)
Comment (rameshmalla) [reply src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt:17]: not required

File: src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt (hunk 2 above)
Comment (rameshmalla) [reply src/main/kotlin/com/tafel/missions/tafelmissions/controller/MissionController.kt:36]: yep

File: src/main/resources/db/migration/V001__CREATE_MISSIONS_TABLE.sql (hunk 3 above)
Comment (rameshmalla) [reply src/main/resources/db/migration/V001__CREATE_MISSIONS_TABLE.sql:18]: Audit had to be designed , but yes we need the updated_by

**Pull request merged**